### DISCUSSION TITLE
Creating a Financial Charts module

### DISCUSSION THREAD

**Discussion by ramkumarkb** [cetz-package/cetz#94] [Ideas]
Hi,

Thanks for the wonderful package.

I would like to create a package for financial charts (something along the lines - https://akela.mendelu.cz/~thala/statcharts/statistical-charts.pdf ). This uses MetaPost and MetaPost has a lot of drawing primitives (e.g. ability to place the labels accurately etc).

Does this package have all the primitives to do such types of charts - import a `json` data, add axes, scale the chart based on the max & min of the input data, draw lables etc. 

I was not sure if I should use the discussion or create an issue in repo.

Thank you !

**Comment by fenjalien** [comment]
Hi, thanks for your interest! 

You have some good timing as we're working on adding a plotting lib #92. We're aiming for it to be similar to tikz's [datavisualization library](https://tikz.dev/dv) and currently debating whether or not it should be kept in CeTZ or be put into its own package/repo.

In terms of primitives we should do? Take a look and if you need something feel free to make a feature request!

**Comment by johannes-wolf** [comment]
Hello :)

this is totally possible, with PR #92 we've got some fundamentals like drawing and scaling axes as functions. Have a look at the PR, I even added `barcharts`, in the new `lib/chart.typ` file. You could use that as foundation to implement others.

**Comment by ramkumarkb** [comment]
@fenjalien & @johannes-wolf  - 

Thank you for the reply. So I will check out the repo and run from the local as the code is just recently merged?

And to read data from a JSON array or object, is there any convinience helper function? May be I will use the Typst default [JSON](https://typst.app/docs/reference/data-loading/json/) and send it to the charts. 

Let me start with some simple Donut charts...

**Reply by johannes-wolf** [reply]
Yes, you can use a local installation.

The chart module can load data from an array of arrays (e.g. CSV) or an array of objects/dicts.
See the `label-key` and `value-key` of the `barchart` function. So depending on the data, default csv or json functions of typst should be enough.
